ethereum/core/src/main/java/org/hyperledger/besu/ethereum/MainnetBlockValidator.java

```diff
@@ -64,64 +63,70 @@ public MainnetBlockValidator(
    *     the block, empty if the block was deemed invalid or couldn't be processed
    */
   @Override
-  public Optional<BlockProcessingOutputs> validateAndProcessBlock(
+  public BlockValidator.Result validateAndProcessBlock(
       final ProtocolContext context,
       final Block block,
       final HeaderValidationMode headerValidationMode,
       final HeaderValidationMode ommerValidationMode) {
+
     final BlockHeader header = block.getHeader();
 
     final MutableBlockchain blockchain = context.getBlockchain();
     final Optional<BlockHeader> maybeParentHeader =
         blockchain.getBlockHeader(header.getParentHash());
     if (maybeParentHeader.isEmpty()) {
-      LOG.error(
-          "Attempted to import block {} with hash {} but parent block {} was not present",
-          header.getNumber(),
-          header.getHash(),
-          header.getParentHash());
+      String errMsg =
+          "Attempted to import block "
+              + block.toLogString()
+              + ", but parent block "
+              + header.getParentHash()
+              + " was not present";
+      LOG.error(errMsg);
       badBlockManager.addBadBlock(block);
-      return Optional.empty();
+      return new Result(errMsg);
     }
     final BlockHeader parentHeader = maybeParentHeader.get();
 
     if (!blockHeaderValidator.validateHeader(header, parentHeader, context, headerValidationMode)) {
       badBlockManager.addBadBlock(block);
-      return Optional.empty();
+      return new Result("Invalid header for block " + block.toLogString());
     }
 
     final Optional<MutableWorldState> maybeWorldState =
         context
             .getWorldStateArchive()
             .getMutable(parentHeader.getStateRoot(), parentHeader.getHash());
     if (!maybeWorldState.isPresent()) {
-      LOG.debug(
-          "Unable to process block {} because parent world state {} is not available",
-          block.getHeader().getNumber(),
-          parentHeader.getStateRoot());
+      String errMsg =
+          "Unable to process block "
+              + block.toLogString()
+              + " because parent world state "
+              + parentHeader.getStateRoot()
+              + " is not available";
+      LOG.debug(errMsg);
       badBlockManager.addBadBlock(block);
-      return Optional.empty();
+      return new Result(errMsg);
     }
     final MutableWorldState worldState = maybeWorldState.get();
 
-    final Result result = processBlock(context, worldState, block);
+    final BlockProcessor.Result result = processBlock(context, worldState, block);
     if (result.isFailed()) {
       badBlockManager.addBadBlock(block);
-      return Optional.empty();
+      return new Result("Error processing block");
```

Comment:
@garyschulte is it enough this generic `Error processing block` message or we should popup more details from `processBlock` method?

Reply:
good enough IMO.  If we need more specifics we can always implement it, but the api we are trying to support is not user facing nor is the consumer of this api user facing.  The only real motivation is better logging IMO

Reply:
same as above